Test/Makefile

```diff
@@ -511,3 +517,95 @@ test_float_in_struct:
+	grep -q "second value = 2.700000" ${TEST_DIR}/SimpleFloatThroughChannel.sim.output
+	grep -q "2 processes created" ${TEST_DIR}/SimpleFloatThroughChannel.sim.output
+	${DELETE_TEST_DIR}
+
+#Test channel with floats enumeration 
+test_channel_with_floats_enumeration:
+	echo "Test channel with floats enumeration"
+	${CREATE_TEST_DIR}
+	cd ${TEST_DIR} && \
+	${SPIN} -run ${FLOAT_CHANNEL_MODEL_DIR}/ChannelsWithFloatsCanBeEnumerated.pml > ChannelsWithFloatsCanBeEnumerated.output
+	cd ${TEST_DIR} && \
+	${SPIN} ${FLOAT_CHANNEL_MODEL_DIR}/ChannelsWithFloatsCanBeEnumerated.pml > ChannelsWithFloatsCanBeEnumerated.sim.output
+	grep -q "State-vector 56 byte, depth reached 17, errors: 0" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.output
+	grep -q "18 states, stored" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.output
+	grep -q "0 states, matched" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.output
+	grep -q "3.140000" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.sim.output
+	grep -q "2.700000" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.sim.output
+	grep -q "2 processes created" ${TEST_DIR}/ChannelsWithFloatsCanBeEnumerated.sim.output
+	${DELETE_TEST_DIR}
+
+#Test complex structures with floats in channels
+test_complex_structures_with_floats_in_channels:
+	echo "Test complex structures with floats in channels"
+	${CREATE_TEST_DIR}
+	cd ${TEST_DIR} && \
+	${SPIN} -run -a ${FLOAT_CHANNEL_MODEL_DIR}/ComplexFloatMessage.pml > ComplexFloatMessage.output
+	cd ${TEST_DIR} && \
+	${SPIN} ${FLOAT_CHANNEL_MODEL_DIR}/ComplexFloatMessage.pml > ComplexFloatMessage.sim.output
+	grep -q "State-vector 404 byte, depth reached 295, errors: 0" ${TEST_DIR}/ComplexFloatMessage.output
+	grep -q "703 states, stored" ${TEST_DIR}/ComplexFloatMessage.output
+	grep -q "154 states, matched" ${TEST_DIR}/ComplexFloatMessage.output
+	grep -q "hash conflicts:         0 (resolved)" ${TEST_DIR}/ComplexFloatMessage.output
+	grep -q "Receiver: index=11, value=0.250000" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "Final message received!" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "queue 1 (queue): \[11,0.250000\]\[11,0.300000\]" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "queue 2 (dqueue): \[0.100000,0.200000,0.300000,0.400000,0.500000,0.600000,0.700000,0.800000,0.900000,0.750000,0,0.000000,52\]\[0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0,5.250000,0\]\[0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,45,0.000000,0\]" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "finalValue = 0.250000" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "finalIndex = 11" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "ComplexFloatMessage.pml:80 (state 25)" ${TEST_DIR}/ComplexFloatMessage.sim.output
+	grep -q "ComplexFloatMessage.pml:51 (state 25) <valid end state>" ${TEST_DIR}/ComplexFloatMessage.sim.output
```

Comment:
I don't think we need to test the end states.

Reply:
in this particular test, the data was intentionally dumped to get all global vars. I've used blocked message sending for that purpose, so, the receiver should go to the valid end state while the sender hangs-up. That is why I verify end states here